R/manip.r

```diff
@@ -12,6 +12,38 @@ mutate.tbl_svy <- function(
     stop("Cannot modify survey variable")
   }
 
+  # Set current_svy so available to svy stat functions
+  old <- set_current_svy(list(full = .data, split = list(.data)))
```

Comment:
`group_split(.data)` is almost* equivalent to `list(.data)`, so I'm going to change this so it only has one method so there's less code.

*It's really equivalent to `list(.data[rep(TRUE, nrow(.data), ]))` because the survey package adds a `popsize` to the fpc attribute when subsetting even when nothingis removed. However, I'm assuming that the math works out so  that these are the same in practice.

``` r
library(srvyr)
library(dplyr)
data(api, package = "survey")

dstrata <- apistrat %>%
  as_survey(strata = stype, weights = pw)


waldo::compare(group_split(dstrata), list(dstrata))
#> `old[[1]]$fpc` is length 1
#> `new[[1]]$fpc` is length 2
#> 
#> `names(old[[1]]$fpc)`: "sampsize"           
#> `names(new[[1]]$fpc)`: "popsize"  "sampsize"
#> 
#> `old[[1]]$fpc$popsize` is absent
#> `new[[1]]$fpc$popsize` is NULL

waldo::compare(group_split(dstrata), list(dstrata[rep(TRUE, nrow(dstrata))]))
#> ✓ No differences
```

<sup>Created on 2022-02-19 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Reply:
And I'm glad I didn't waste  your time with this - using group_split made some weirdness happen for twophase and db-backed surveys.

Reply:
Great- thanks for getting this over the finish line! For `filter()`, it just totally slipped my mind. Thanks for adding that to the PR.